src/java/org/apache/cassandra/index/sai/disk/v1/VectorIndexSearcher.java

```diff
@@ -76,8 +73,32 @@ public KeyRangeIterator search(Expression exp, QueryContext context, int limit)
 
         ByteBuffer buffer = exp.lower.value.raw;
         float[] queryVector = (float[])indexContext.getValidator().getSerializer().deserialize(buffer.duplicate());
+        var results = graph.search(queryVector, limit, null, Integer.MAX_VALUE);
```

Comment:
partition-restricted query is not supported yet because `Bits` is not implemented and `AnnPostingList#advance` will skip over restricted primary key's row id.

We can leave it to separate PR.

Reply:
what code is normally responsible for restricting to a single partition?

Reply:
It's in `ResultRetriever#computeNext`.

```
            // If being called for the first time, skip to the beginning of the range.
            // We can't put this code in the constructor because it may throw and the caller
            // may not be prepared for that.
            if (lastKey == null)
                resultKeyIterator.skipTo(firstPrimaryKey);
```

`skipTo` will eventually call `PostingList#advance`

Reply:
Thanks, I made a note to tackle this

Getting a Jira board created now so it's not just a gdoc...